neo/rawio/neuralynxrawio/neuralynxrawio.py

```diff
@@ -128,32 +131,68 @@ class NeuralynxRawIO(BaseRawIO):
         ("channel_id", "uint32"),
         ("sample_rate", "uint32"),
         ("nb_valid", "uint32"),
-        ("samples", "int16", (NcsSection._RECORD_SIZE)),
+        ("samples", "int16", NcsSection._RECORD_SIZE),
     ]
 
     def __init__(
-        self, dirname="", filename="", exclude_filename=None, keep_original_times=False, strict_gap_mode=True, **kargs
+            self,
+            dirname="",
+            include_filenames=None,
+            exclude_filenames=None,
+            keep_original_times=False,
+            strict_gap_mode=True,
+            filename=None,
+            exclude_filename=None,
+            **kargs
     ):
 
-        if dirname != "":
-            self.dirname = dirname
-            self.rawmode = "one-dir"
-        elif filename != "":
-            self.filename = filename
-            self.rawmode = "one-file"
+        if not dirname:
+            raise ValueError("`dirname` cannot be empty.")
+
+        if filename:
+            include_filenames = filename
+            warnings.warn("`filename` is deprecated and will be removed. Please use `include_filenames` instead")
+
+        if exclude_filename:
+            exclude_filenames = exclude_filename
+            warnings.warn("`exclude_filename` is deprecated and will be removed. Please use `exclude_filenames` instead")
+
+        if include_filenames is None:
+            include_filenames = []
+        elif not isinstance(include_filenames, (list, set, np.ndarray)):
+            include_filenames = [include_filenames]
```

Comment:
I don't know if we should allow this case. include_filenames should be an Iterable. I think I tuple would be more common than a set. I would probably say just set it equal and then let it error later if the user does it wrong. So I would do

```python
if include_filenames is None:
    include_filenames = []
```